src/components/charts/helper.js

```diff
@@ -47,29 +47,47 @@ const getCategoryColorFrom2Colors = index => colorsForCharts.categorical2Colors[
 
 const getDataKey = showPercentage => (showPercentage ? 'percentage' : 'value');
 
-const transformArrangerDataToChart = (field) => {
+const transformArrangerDataToChart = (field, sqonValues) => {
   const chartData = [];
-  field.buckets.map(bucket =>
-    chartData.push({
-      name: bucket.key,
-      value: bucket.doc_count,
-    }),
-  );
+  field.buckets
+    .filter(bucket => (sqonValues === undefined || sqonValues.includes(bucket.key)))
+    .forEach(bucket =>
+      chartData.push({
+        name: bucket.key,
+        value: bucket.doc_count,
+      }),
+    );
   return chartData;
 };
 
-const transformArrangerDataToSummary = (field, chartType, title) => ({
+const transformArrangerDataToSummary = (field, chartType, title, sqonValues) => ({
   type: chartType,
   title,
-  data: transformArrangerDataToChart(field),
+  data: transformArrangerDataToChart(field, sqonValues),
 });
 
-const transformDataToCount = (field, label) => ({
+const transformDataToCount = (field, label, sqonCount) => ({
   label,
-  value: field.buckets.length,
+  value: Math.min(field.buckets.length, sqonCount),
```

Comment:
When there are no selected values in a given `field`, this means all values should be counted, so here I think it would be easy to understand that we let `sqonCount = Inifinity`. That's why here I use `Math.min`. I can change it to a `if...else` statement if you want.